### PR TITLE
ER-1101: Show only published nodes in feed and carousels

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/ER-1101

### CODE REVIEW

**Comment by cableman** [comment]
Could this not be handle via permissions? If the user is not allowed to se unpublished content the nodes should not be loaded in `node_load`.

File: sites/all/modules/reol_app_feeds/src/Helper/NodeHelper.php [hunk 1]
```diff
@@ -230,7 +230,7 @@ public function loadReferences($entity, $field_name, $multiple = TRUE) {
     }
     $values = $entity->{$field_name}[LANGUAGE_NONE];
     $nids = array_column($values, 'target_id');
-    $nodes = node_load_multiple($nids);
+    $nodes = $this->loadNodes($nids);
```
Comment (cableman) [inline sites/all/modules/reol_app_feeds/src/Helper/NodeHelper.php:233]: `entity_load` has an `$conditions` array that should be used, where the condition could be node status.

**Review by cableman — CHANGES_REQUESTED** [review]

**Comment by rimi-itk** [comment]
`node_load_multiple` load all requested nodes (including unpublished ones). How can we use permissions to make it not load unpublished nodes and not run into weird caching issues?

**Comment by cableman** [comment]
Is for removal in D8, but this site an code is never going to be moved outside d7. But Still do not understand why the permissions do not prevent users for seeing un-published content?

**Comment by rimi-itk** [comment]
Users cannot see unpublished content (on `/node/…`), but I don't know how/if we can make `node_load_multiple` check permissions.

**Comment by cableman** [comment]
That is strange, it seams that node_load is not checking permissions and the way to go it to only give it id's for the nodes that should be allowed to view. But can't see how to do that when you only have an id.... without loading the node.... f... it

**Review by cableman — APPROVED** [review]

**Pull request merged**